lib/route.js

```diff
@@ -344,6 +345,15 @@ function buildRouting (options) {
       }
     }
 
+    const connHeader = String.prototype.toLowerCase.call(req.headers.connection || '')
+    if (connHeader === 'keep-alive') {
+      keepAliveConnections.add(req.socket)
+      req.socket.on('close', removeTrackedSocket.bind({ keepAliveConnections, socket: req.socket }))
+      function removeTrackedSocket() {
+        this.keepAliveConnections.delete(this.socket)
+      }
```

Comment:
Move this to a top level function. You are preventing all previous requests from being garbage collected in this way.

Reply:
The bind is meant to combat that.